linera-base/src/crypto/secp256k1.rs

```diff
@@ -34,17 +51,42 @@ pub struct Secp256k1KeyPair {
 
 /// A secp256k1 signature.
 #[derive(Eq, PartialEq, Copy, Clone)]
-pub struct Secp256k1Signature(pub secp256k1::ecdsa::Signature);
+pub struct Secp256k1Signature(pub Signature);
 
 impl Secp256k1PublicKey {
     /// A fake public key used for testing.
     #[cfg(with_testing)]
     pub fn test_key(seed: u8) -> Self {
         use rand::SeedableRng;
-
         let mut rng = rand::rngs::StdRng::seed_from_u64(seed as u64);
-        let secp = secp256k1::Secp256k1::signing_only();
-        Self(secp.generate_keypair(&mut rng).1)
+        let sk = k256::SecretKey::random(&mut rng);
+        Self(sk.public_key().into())
+    }
+
+    /// Returns the bytes of the public key in compressed representation.
+    pub fn as_bytes(&self) -> Vec<u8> {
```

Comment:
Should this be a `[u8; 33]`?

Reply:
The conversion is fallible but I guess it should work since we're operating on a valid key here.

Reply:
Yes, and the type guards its invariant, right? Both in all constructors and on deserialization?

Reply:
Not necessarily. This is a compressed representation - 33 bytes - and the uncompressed one can be 65 bytes IIRC so - in theory - this conversion could fail if we ever switched to uncompressed representation. But there's a unit test for that.

Reply:
Done in [Apply review suggestions.](https://github.com/linera-io/linera-protocol/pull/3407/commits/8f530c26831f602c5a89abc74abc7445a02389d6).